services/base-jdbc/src/main/java/org/eclipse/hono/service/base/jdbc/config/JdbcProperties.java

```diff
@@ -140,4 +179,16 @@ public static JDBCClient dataSource(final Vertx vertx, final JdbcProperties data
 
     }
 
+    private static void putValidValueIntoConfig(final JsonObject config, final String label, final int value, final int limit, final boolean checkLowerLimit) {
```

Comment:
Given that we have defined the limits at the class level, by means of constants, I believe that we should enforce these limits in the setters instead of deferring it to the time the datasource is being constructed (which may actually never happen), or am I mistaken?

Reply:
For the non negativ checks I think you are right, we could do those checks in the setter methods. However, for the initialPoolSize we would have the problem that we don't have control over the order in which someone calls the setter methods. So for example if you call setInitialPoolSize with input 5, which is between the defaults for min and maxPoolSize, this would be seen as valid input and initialPoolSize would be set to 5. After that you could just change the maxPoolSize to be below 5 or the minPoolSize to be above 5, which would result in an invalid configuration. Therefor, in my opinion it makes more sense to enforce this right before we create the JDBCClient, at which point we can be sure all the values have been set. What do you think?

Reply:
Ok, I see your point